docs/messaging/azure-web-pubsub-integration.md

```diff
@@ -2,135 +2,305 @@
+  - A tag is added to the Web PubSub resource with a key of `ExampleKey` and a value of `Example value`.
+
+There are many more configuration options available to customize the Web PubSub resource resource. For more information, see <xref:Azure.Provisioning.PostgreSql>. For more information, see [`Azure.Provisioning` customization](../azure/integrations-overview.md#azureprovisioning-customization).
+
+### Connect to an existing Azure Web PubSub instance
+
+You might have an existing Azure Web PubSub service that you want to connect to. You can chain a call to annotate that your <xref:Aspire.Hosting.ApplicationModel.AzureWebPubSubResource> is an existing resource:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var existingPubSubName = builder.AddParameter("existingPubSubName");
+var existingPubSubResourceGroup = builder.AddParameter("existingPubSubResourceGroup");
+
+var webPubSub = builder.AddAzureWebPubSub("web-pubsub")
+                       .AsExisting(existingPubSubName, existingPubSubResourceGroup);
+
+builder.AddProject<Projects.ExampleProject>()
+       .WithReference(webPubSub);
+
+// After adding all resources, run the app...
+```
+
+For more information on treating Azure Web PubSub resources as existing resources, see [Use existing Azure resources](../azure/integrations-overview.md#use-existing-azure-resources).
+
+Alternatively, instead of representing an Azure Web PubSub resource, you can add a connection string to the app host. Which is a weakly-typed approach that's based solely on a `string` value. To add a connection to an existing Azure Web PubSub service, call the <xref:Aspire.Hosting.ParameterResourceBuilderExtensions.AddConnectionString%2A> method:
+
+```csharp
+var builder = DistributedApplication.CreateBuilder(args);
+
+var webPubSub = builder.ExecutionContext.IsPublishMode
+    ? builder.AddAzureWebPubSub("web-pubsub")
+    : builder.AddConnectionString("web-pubsub");
+
+builder.AddProject<Projects.ExampleProject>()
+       .WithReference(webPubSub);
+
+// After adding all resources, run the app...
+```
+
+[!INCLUDE [connection-strings-alert](../includes/connection-strings-alert.md)]
+
+The connection string is configured in the app host's configuration, typically under User Secrets, under the `ConnectionStrings` section:
+
+```json
 {
-    // Use client...
+  "ConnectionStrings": {
+    "web-pubsub": "https://{account_name}.webpubsub.azure.com"
+  }
 }
 ```
 
-For more information, see the [Azure.Messaging.WebPubSub documentation](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/webpubsub/Azure.Messaging.WebPubSub/README.md).
+For more information, see [Add existing Azure resources with connection strings](../azure/integrations-overview.md#add-existing-azure-resources-with-connection-strings).
 
-## App host usage
+## Client integration
```

Comment:
How about ServiceClient? In WebPubSub world, there is also a real "client" role which is the WebPubSub clients. Here I believe it talks about the "server" side. Using `client` might lead to confusion here. Same concern applies to below sections.

![image](https://github.com/user-attachments/assets/bcf17b1b-4a5b-455f-b950-49a9bdfa249a)

Reply:
➕1️⃣